src/sinks/influxdb_metrics.rs

```diff
@@ -0,0 +1,916 @@
+                        tags,
+                        Some(fields),
+                        ts,
+                    )])
+                }
+                MetricValue::Distribution {
+                    values,
+                    sample_rates,
+                } => {
+                    let fields = encode_distribution(&values, &sample_rates);
+                    Some(vec![influx_line_protocol(
+                        fullname,
+                        "distribution",
+                        tags,
+                        fields,
+                        ts,
+                    )])
+                }
+            }
+        })
+        .flatten()
+        .filter(|lp| !lp.is_empty())
+        .collect()
+}
+
+fn encode_distribution(values: &[f64], counts: &[u32]) -> Option<HashMap<String, Field>> {
+    if values.len() != counts.len() {
+        return None;
+    }
+
+    let mut samples = Vec::new();
+    for (v, c) in values.iter().zip(counts.iter()) {
+        for _ in 0..*c {
+            samples.push(*v);
+        }
+    }
+
+    if samples.is_empty() {
+        return None;
+    }
+
+    if samples.len() == 1 {
+        let val = samples[0];
+        return Some(
+            vec![
+                ("min".to_owned(), Field::Float(val)),
+                ("max".to_owned(), Field::Float(val)),
+                ("median".to_owned(), Field::Float(val)),
+                ("avg".to_owned(), Field::Float(val)),
+                ("sum".to_owned(), Field::Float(val)),
+                ("count".to_owned(), Field::Float(1.0)),
+                ("quantile_0.95".to_owned(), Field::Float(val)),
+            ]
+            .into_iter()
+            .collect(),
+        );
+    }
+
+    samples.sort_by(|a, b| a.partial_cmp(b).unwrap_or(Ordering::Equal));
+
+    let length = samples.len() as f64;
+    let min = samples.first().unwrap();
+    let max = samples.last().unwrap();
+
+    let p50 = samples[(0.50 * length - 1.0).round() as usize];
+    let p95 = samples[(0.95 * length - 1.0).round() as usize];
+
+    let sum = samples.iter().sum();
+    let avg = sum / length;
+
+    let fields: HashMap<String, Field> = vec![
+        ("min".to_owned(), Field::Float(*min)),
+        ("max".to_owned(), Field::Float(*max)),
+        ("median".to_owned(), Field::Float(p50)),
+        ("avg".to_owned(), Field::Float(avg)),
+        ("sum".to_owned(), Field::Float(sum)),
+        ("count".to_owned(), Field::Float(length)),
+        ("quantile_0.95".to_owned(), Field::Float(p95)),
+    ]
+    .into_iter()
+    .collect();
+
+    Some(fields)
+}
+
+// https://v2.docs.influxdata.com/v2.0/reference/syntax/line-protocol/
+fn influx_line_protocol(
+    measurement: String,
+    metric_type: &str,
+    tags: Option<HashMap<String, String>>,
+    fields: Option<HashMap<String, Field>>,
+    timestamp: i64,
+) -> String {
+    let mut line_protocol = vec![encode_key(measurement)];
+
+    // Tags
+    let mut unwrapped_tags = tags.unwrap_or(HashMap::new());
```

Comment:
Using [`unwrap_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_else) instead of `unwrap_or` would create `HashMap` lazily, only when it is actually necessary.

Reply:
Done